File: Jan2021/01Jan/CanPlaceFlowers.cpp
/*
Problem name: 605. Can Place Flowers
Problem link: https://leetcode.com/problems/can-place-flowers/
*/

class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        for(int i=0;i<flowerbed.size();++i){
            if(n==0)
                break;
            if(!flowerbed[i]){
                if (i > 0 && flowerbed[i-1]) continue;
                if (i < flowerbed.size() - 1 && flowerbed[i + 1]) continue;
                flowerbed[i]=1;
                n--;
            }
        }
        if(!n)
            return true;
        else
            return false;
    }
};

File: Jan2021/1Jan/CanPlaceFlowers.java
/*
Problem Name: 605. Can Place Flowers
Problem Link: https://leetcode.com/problems/can-place-flowers/
*/
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int cnt = 0, len = flowerbed.length, i;
        for(i=0;i<len && flowerbed[i]==0;i++,cnt++);
        if(i == len)
            n -= (cnt+1)/2;
        else
            n-= cnt/2;
        cnt = 0;
        for(;i<len;i++){
            if(flowerbed[i] == 0)
                cnt++;
            else{
                n-= (cnt-1)/2;
                cnt = 0;
            }
        }
        n -= cnt/2;
        return n <= 0;
    }
}
